### PR TITLE
databases-week1

### PR DESCRIPTION
**Pull request by Yuliia-Balandiuk**

### CODE REVIEW

File: databases/week1/hw-week1.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+WHERE `name` NOT REGEXP 'done';
+
+-- 5 Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM `task`
+ORDER BY `created` DESC;
+
+-- 6 Get the single most recently created task
+SELECT MAX(created) 
+FROM `task`;
```
Comment (Elia-Git) [inline databases/week1/hw-week1.sql]: Here no one knows which one of the tasks is that(at least task id needed)
Try this if it makes sense
```
SELECT id,title,created
FROM task
ORDER BY created DESC LIMIT 1;
```

File: databases/week1/hw-week1.sql [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+-- 7 Get the title and due date of all tasks where the title or description contains database
+SELECT `title`, `due_date`
+FROM `task`
+WHERE `title` REGEXP 'database' OR 
+      `description` REGEXP 'database';
```
Comment (Elia-Git) [inline databases/week1/hw-week1.sql]: A small fix with id ..
```
SELECT `id` ,`title`, `due_date`
FROM `task`
WHERE `title` REGEXP 'database' OR 
      `description` REGEXP 'database';
```

File: databases/week1/hw-week1.sql [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+-- 8 Get the title and status (as text) of all tasks
+SELECT `title`, status.name AS text
+FROM `task`
+JOIN status ON task.status_id = status.id;
+
+-- 9 Get the name of each status, along with a count of how many tasks have that status
+SELECT status.name AS text, COUNT(*) as `amount`
+FROM `task`
+JOIN status ON task.status_id = status.id
+GROUP BY status.name;
```
Comment (Elia-Git) [inline databases/week1/hw-week1.sql:30]: I would also add  task `id` (unique field) too.

File: databases/week1/DB-week1.sql [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  `created` DATETIME NOT NULL,
+  `updated` DATETIME NOT NULL,
+  `due_date` DATETIME NULL DEFAULT NULL,
+  `status_id` int(10) unsigned NOT NULL,
+  `user_id` int(10) unsigned,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_status` FOREIGN KEY (`status_id`) REFERENCES `status` (`id`) ON DELETE CASCADE,
+  CONSTRAINT `fk_user` FOREIGN KEY (`user_id`) REFERENCES `user` (`id`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
```
Comment (Elia-Git) [inline databases/week1/DB-week1.sql]: Committing empty or commented lines is not a good practice. Don't commit gaps/empty lines or commented lines.

**Review by Elia-Git — APPROVED** [review]
Good job in general. Pls, ignore the empty code of lines (lines..4,9,15 & soon)  in your commit for better practice and readability.
You can merge it!

File: databases/week1/hw-week1.sql (hunk 1 above)
Comment (Yuliia-Balandiuk) [reply databases/week1/hw-week1.sql]: Thank you for your review. I agree with you, it looks better.

**Pull request merged**